lib/package.rb

```diff
@@ -129,6 +129,7 @@ def self.system(*args, **opt_args)
     else
       env = CREW_ENV_OPTIONS_HASH
     end
+    if CREW_NO_ADD_ENV_TO_MAKE then env = { "CREW_NO_ADD_ENV_TO_MAKE " => '1' } end
```

Comment:
`env` is not a key word?  Also, how is this applied?  Don't we need an update to crew also?

Reply:
I can't speak to it being a key word, but I see `env` being used thus in `package.rb`:

```
# extract env hash
    if args[0].is_a?(Hash)
      env = CREW_ENV_OPTIONS_HASH.merge(args[0])
      args.delete_at(0) # remove env hash from args array
    else
      env = CREW_ENV_OPTIONS_HASH
    end
    if CREW_NO_ADD_ENV_TO_MAKE then env = { "CREW_NO_ADD_ENV_TO_MAKE " => '1' } end
```
and also here:
```
begin
      Kernel.system(env, cmd_args, **opt_args)
    rescue => e
      exitstatus = $?.exitstatus
      # print failed line number and error message
      puts "#{e.backtrace[1]}: #{e.message}".orange
      raise InstallError, "`#{env} #{cmd_args}` exited with #{exitstatus}"
    end
```

No update to crew is needed. We got rid of crew having to be updated for new constants some time back. Just `const.rb` needs to be updated.

I believe the way to have this applied is thus:
```
  def self.build
    ENV['CREW_NO_ADD_ENV_TO_MAKE'] = '1'
    warn_level = $VERBOSE
    $VERBOSE = nil
    load "#{CREW_LIB_PATH}lib/const.rb"
    $VERBOSE = warn_level
    FileUtils.mkdir('builddir')
    Dir.chdir('builddir') do
      system "#{@musldep_cmake_options} \
      -DOPENSSL_INCLUDE_DIR=#{CREW_MUSL_PREFIX}/include \
      -DCARES_STATIC=ON \
      -DCARES_STATIC_PIC=ON \
      -DCARES_BUILD_TESTS=ON \
      -DCARES_BUILD_CONTAINER_TESTS=ON \
      -DCARES_BUILD_TOOLS=ON \
      -DCARES_SHARED=OFF \
      ../ -G Ninja"
    end
    system 'samu -C builddir'
  end
```

or 
```
  def self.build
    ENV['CREW_NO_ADD_ENV_TO_MAKE'] = '1'
    warn_level = $VERBOSE
    $VERBOSE = nil
    load "#{CREW_LIB_PATH}lib/const.rb"
    $VERBOSE = warn_level
    system 'make'
  end
```

Reply:
Am I missing something?  I don't see where this takes effect if set.

Reply:
Try a package file with 
```
def self.build
    ENV['CREW_NO_ADD_ENV_TO_MAKE'] = '1'
    warn_level = $VERBOSE
    $VERBOSE = nil
    load "#{CREW_LIB_PATH}lib/const.rb"
    $VERBOSE = warn_level
    system 'exit 1'
  end
```

You should see only `CREW_NO_ADD_ENV_TO_MAKE=1` being set in the error output,instead of having all of `CREW_ENV_OPTIONS` in there...

Reply:
Line 132 is where `env` is redefined to be a nop if the env variable exists.

Reply:
```
crew build test
test: Test Package.
Downloading source...
Building from source, this may take a while...
/usr/local/lib/crew/lib/package.rb:141:in `system': Command failed with exit 1: exit 1
test failed to build: `{"CREW_NO_ADD_ENV_TO_MAKE "=>"1"} exit 1` exited with 1
```

Reply:
test package: `test.rb`
```
require 'package'

class Test < Package
  description 'Test Package.'
  homepage 'http://chromebrew'
  @_ver = '1.0'
  version @_ver
  license 'BSD and GPL-2'
  compatibility 'all'
  source_url 'SKIP'


  def self.build
   ENV['CREW_NO_ADD_ENV_TO_MAKE'] = '1'
    warn_level = $VERBOSE
    $VERBOSE = nil
    load "#{CREW_LIB_PATH}lib/const.rb"
    $VERBOSE = warn_level
    system "exit 1"
  end

  def self.check
    ENV['CREW_NO_ADD_ENV_TO_MAKE'] = '1'
    warn_level = $VERBOSE
    $VERBOSE = nil
    load "#{CREW_LIB_PATH}lib/const.rb"
    $VERBOSE = warn_level
    system 'make', 'test'
  end

  def self.install
    ENV['CREW_NO_ADD_ENV_TO_MAKE'] = '1'
    warn_level = $VERBOSE
    $VERBOSE = nil
    load "#{CREW_LIB_PATH}lib/const.rb"
    $VERBOSE = warn_level
    system "make DESTDIR=#{CREW_DEST_DIR} install"
  end
end
```

Reply:
I'm not printing out the hash in the error message command line output, but open to suggestions on how to do that best...

Reply:
Oh, so you are just overriding the `env` value?  I'm thinking it might be better to just not set `env` in the first place.  That would require a change to `crew`, however.

Reply:
I was trying to have a light touch. `env` needs to be a hash, and it's being used for every `system` line via that mechanism at the end of `package.rb`.  So it seemed best to just set the hash to something minimal which indicated what had been done.

Reply:
But if you think that it is better to edit `crew` feel free to offer alternative code?

Reply:
It would be easier to understand what is going on if the logic was all in one place.  ie. Add a conditional to set based on the const value.

Reply:
Where would you put it though in crew? `system` could be called from anywhere.

Maybe we could add a comment line right above `if CREW_NO_ADD_ENV_TO_MAKE then env = { "CREW_NO_ADD_ENV_TO_MAKE " => '1' } end`?

Reply:
Sorry, I said `crew` earlier but I should have said `const.rb`.
```
CREW_DISABLE_ENV_OPTIONS = ENV['CREW_DISABLE_ENV_OPTIONS']
CREW_ENV_OPTIONS_HASH = {                                                                                                                                                                 
  'CFLAGS'   => CREW_COMMON_FLAGS,                                                                                                                                                        
  'CXXFLAGS' => CREW_COMMON_FLAGS,                                                                                                                                                        
  'FCFLAGS'  => CREW_COMMON_FLAGS,                                                                                                                                                        
  'FFLAGS'   => CREW_COMMON_FLAGS,                                                                                                                                                        
  'LDFLAGS'  => CREW_LDFLAGS                                                                                                                                                              
}                                                                                                                                                                                         
# parse from hash to shell readable string                                                                                                                                                
CREW_ENV_OPTIONS = CREW_DISABLE_ENV_OPTIONS ? 'CREW_DISABLE_ENV_OPTIONS=1' : CREW_ENV_OPTIONS_HASH.map {|k, v| "#{k}=\"#{v}\"" } .join(' ')
```